### PR TITLE
Convert CoreObject classes to ES6 classes extending CoreObject

### PR DESCRIPTION
**Pull request by Turbo87**

### CODE REVIEW

File: lib/models/addon-discovery.js [hunk 1]
```diff
@@ -21,11 +20,10 @@ const heimdall = require('heimdalljs');
   @extends CoreObject
   @constructor
 */
-let AddonDiscovery = CoreObject.extend({
-  init(ui) {
-    this._super.apply(this, arguments);
+class AddonDiscovery {
```
Comment (nathanhammond) [inline lib/models/addon-discovery.js]: Don't we need to extend `CoreObject` here? Or are we calling this internal-only?

File: lib/models/addons-factory.js [hunk 2]
```diff
@@ -16,13 +15,11 @@ const heimdall = require('heimdalljs');
   @extends CoreObject
   @constructor
 */
-let AddonsFactory = CoreObject.extend({
-  init(addonParent, project) {
-    this._super();
-
+class AddonsFactory {
```
Comment (nathanhammond) [inline lib/models/addons-factory.js]: (Also here.)

File: lib/models/addon-discovery.js (hunk 1 above)
Comment (Turbo87) [inline lib/models/addon-discovery.js]: > are we calling this internal-only?

👍

**Comment by stefanpenner** [comment]
yay nice things!

**Pull request merged**